schema/resolvers.js

```diff
@@ -176,7 +209,83 @@ const Resolvers = {
     section: ({ sectionId }, args, ctx) => {
       return ctx.repositories.Section.get(sectionId);
     },
-    position: (page, args, ctx) => Resolvers.Page.position(page, args, ctx)
+    position: (page, args, ctx) => {
+      return Resolvers.Page.position(page, args, ctx);
+    },
+    routingRuleSet: ({ id }, args, ctx) => {
+      return ctx.repositories.Routing.findRoutingRuleSetByQuestionPageId({
+        QuestionPageId: id
+      });
+    },
+    titleNew: page => page.title
+  },
+
+  RoutingRuleSet: {
+    routingRules: ({ id }, args, ctx) => {
+      return ctx.repositories.Routing.findAllRoutingRules({
+        RoutingRuleSetId: id
+      });
+    },
+    questionPage: ({ questionPageId }, args, ctx) => {
+      return ctx.repositories.Page.get(questionPageId);
+    },
+
+    else: ({ sectionDestination, pageDestination }, args, ctx) => {
+      if (isNil(sectionDestination)) {
+        return null;
+      } else if (isNil(pageDestination)) {
+        return ctx.repositories.Section.get(sectionDestination);
+      } else {
+        return ctx.repositories.Page.get(pageDestination);
+      }
+    }
+  },
+
+  RoutingRule: {
+    conditions: ({ id }, args, ctx) => {
+      return ctx.repositories.Routing.findAllRoutingConditions({
+        RoutingRuleId: id
+      });
+    },
+    goto: ({ sectionDestination, pageDestination }, args, ctx) => {
+      if (isNil(sectionDestination)) {
+        return null;
+      } else if (isNil(pageDestination)) {
+        return ctx.repositories.Section.get(sectionDestination);
+      } else {
+        return ctx.repositories.Page.get(pageDestination);
+      }
```

Comment:
Yeah its a bit of a smell. For now, I can extract the logic into one place. We can revisit RoutingDestination again. Semantically a routing destination is either a Section or a Page, and can optionally not be set, so the logic is correct.

There was some debate about whether RoutingDestination should be a type in its own right. We opted against it because its really nothing more than a join table between routingCondition, routingRuleSet, section and questionPage. We can always re-evaluate if it makes sense to migrate to this in future.